frontend/src/Editor/Editor.jsx

```diff
@@ -821,6 +827,36 @@ class EditorComponent extends React.Component {
     this.setState({ showHiddenOptionsForDataQueryId: dataQueryId });
   };
 
+  createInputFieldToRenameQuery = (id) => {
+    this.renameQueryNameId.current = id;
+    this.setState({ renameQueryName: true });
+  };
+
+  updateQueryName = (selectedQueryId, newName) => {
+    if (newName) {
```

Comment:
Yes, if queryName is not changed, then newName and query name are equal. Adding one more condition to avoid updating queryName in such a scenario.
`if(newName && newName !== this.state.selectedQuery.name)`